### DISCUSSION TITLE
Reset password flow

### DISCUSSION THREAD

**Discussion by ByteSizeSkaffy** [supabase/supabase#11478] [Questions] [answered]
``What I'm trying to do is reset the password of a user through email, 
though when I'm redirected and try updating the user (when you go through the email redirect link it should sign in the user right?)
I get the error 'supabase.auth.updateUser is not a function' 
which upon further investigation seems to come from the user object being null.
I'm using supabase version 1 for reasons beyond my controlm I'm also programming in Vue
Code of the password alteration menu:
```
<script>
import { ref } from "vue";
//import { uid } from "uid";
import {supabase} from "../supabase/init";

export default {
  name: "Alterpassword",
  setup() {
    const password1 = ref("");
    const password2 = ref("");
    var errorMsg = ref("");
    const user = async ()=> supabase.auth.user();
    const validPassword = ()=> (password1.value==password2.value);
    const updateUser = async ()=> {
    if(validPassword){
      console.log("user = " + user.error )
      try{
        console.log("validpassword")
        const {  data, error } = await supabase.auth.updateUser({password: password1.value});
        console.log(error, data);
      }catch(error){
        console.log(error);
        errorMsg.value = error.message;
      }finally{
        errorMsg.value = "Your password has been changed";
        console.log("finally")        
      }
    }
    }; 
    return {
      updateUser,      
    }
    }
    };
</script>
```

**Comment by silentworks** [comment, marked as answer]
You are using the wrong method if you are on v1, you can checkout the docs for v1 to see you need `supabase.auth.update({ password: new_password })`. https://supabase.com/docs/reference/javascript/v1/auth-update
